django/api/services/bcngws.py

```diff
@@ -0,0 +1,51 @@
+import requests
+from django.conf import settings
+
+def get_placename_matches(names_list):
```

Comment:
I think it might be cleaner to make this a recursive function, something like:

```
def get_placename_matches(names_list, page_size, start_index, result):
         # do a call to the api here, and get the response.
         # filter the response by the community names
         # then refined_result = result + filteredResponse.result (this is pseudocode)
         # then if response.totalResults >= start_index + page_size, call get_placename_matches()
         # with the same names_list, same page_size, start_index = start_index + page_size, refined_result
         # if response.totalResults < start_index + page_size, return refined_result
```

The initial call to get_placename_matches should use args (names_list, 200, 1, [ ] )